File: source/bindings/java/com/microsoft/cognitiveservices/speech/KeywordRecognitionEventArgs.java
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
package com.microsoft.cognitiveservices.speech;

import com.microsoft.cognitiveservices.speech.PropertyId;
import com.microsoft.cognitiveservices.speech.util.Contracts;
import com.microsoft.cognitiveservices.speech.util.IntRef;

/**
 * Defines content of an keyword recognizing/recognized events.
 */
public class KeywordRecognitionEventArgs extends com.microsoft.cognitiveservices.speech.RecognitionEventArgs {

    /*! \cond INTERNAL */

    /**
     * Constructs an instance of a KeywordRecognitionEventArgs object.
     * @param eventArgs recognition event args object.
     */
    KeywordRecognitionEventArgs(long eventArgs) {
        super(eventArgs);
        storeEventData(false);

    }
    
    KeywordRecognitionEventArgs(long eventArgs, boolean dispose) {
        super(eventArgs);
        storeEventData(dispose);
    }

    /*! \endcond */

    /**
     * Represents the keyword recognition result.
     * @return Represents the keyword recognition result.
     */
    public final KeywordRecognitionResult getResult() {
        return result;
    }

    /**
     * Returns a String that represents the session id and the keyword recognition result event.
     * @return A String that represents the keyword recognition result event.
     */
    @Override
    public String toString() {
        return "SessionId:" + getSessionId() +
                " ResultId:" + result.getResultId() +
                " Reason:" + result.getReason() +
                "> Recognized text:<" + result.getText() + ">.";
    }

    private void storeEventData(boolean disposeNativeResources) {
        Contracts.throwIfNull(eventHandle, "eventHandle");
        IntRef resultHandle = new IntRef(0);
        Contracts.throwIfFail(getRecognitionResult(eventHandle, resultHandle));
        this.result = new KeywordRecognitionResult(resultHandle.getValue());
        Contracts.throwIfNull(this.getSessionId(), "SessionId");
        if (disposeNativeResources == true) {
            super.close();
        }
    }
    
    private KeywordRecognitionResult result;
}

File: source/bindings/csharp/keyword_recognition_event_args.cs
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
using System;
using System.Globalization;
using Microsoft.CognitiveServices.Speech.Internal;
using static Microsoft.CognitiveServices.Speech.Internal.SpxExceptionThrower;

namespace Microsoft.CognitiveServices.Speech
{
    /// <summary>
    /// Class for the events emitted by the <see cref="KeywordRecognizer" />.
    /// </summary>
    public class KeywordRecognitionEventArgs: RecognitionEventArgs
    {
        internal KeywordRecognitionEventArgs(IntPtr eventHandlePtr): base(eventHandlePtr)
        {
            IntPtr result = IntPtr.Zero;
            ThrowIfFail(Internal.Recognizer.recognizer_recognition_event_get_result(eventHandle, out result));
            Result = new KeywordRecognitionResult(result);
        }

        /// <summary>
        /// Keyword recognition event result.
        /// </summary>
        public KeywordRecognitionResult Result { get; }

        /// <summary>
        /// Returns a string that represents the keyword recognition result event.
        /// </summary>
        /// <returns>A string that represents the keyword recognition result event.</returns>
        public override string ToString()
        {
            return string.Format(CultureInfo.InvariantCulture,"SessionId:{0} ResultId:{1} Reason:{2} Recognized text:<{3}>.", SessionId, Result.ResultId, Result.Reason, Result.Text);
        }
    }
}
